src/sage/data_structures/stream.py

```diff
@@ -1538,13 +1543,10 @@ def get_coefficient(self, n):
             sage: [h.get_coefficient(i) for i in range(10)]
             [0, 0, 1, 6, 20, 50, 105, 196, 336, 540]
         """
-        c = ZZ.zero()
-        for k in range(self._left._approximate_order,
-                       n - self._right._approximate_order + 1):
-            val = self._left[k]
-            if val:
-                c += val * self._right[n-k]
-        return c
+        return sum(l * self._right[n - k]
```

Comment:
```suggestion
        return ZZ.sum(l * self._right[n - k]
```
A slight difference, but it prevents `int(0)` from being returned. Well, perhaps that is fine?